src/managers/kyuubi.py

```diff
@@ -32,14 +32,49 @@ def update(
         zookeeper_info: ZookeeperInfo | None,
     ):
         """Update Kyuubi service and restart it."""
-        spark_config = SparkConfig(
+        restart_required = False
+
+        new_spark_config = SparkConfig(
             s3_info=s3_info, service_account_info=service_account_info
         ).contents
-        hive_config = HiveConfig(db_info=metastore_db_info).contents
-        kyuubi_config = KyuubiConfig(db_info=auth_db_info, zookeeper_info=zookeeper_info).contents
+        try:
```

Comment:
Improving the way we update the configuration files inside the workload, we now attempt to update the configuration files and restart the container only if something has changed in the configuration from what is already inside the container. This will greatly improve the performance when `config-changed` event is triggered multiple times.